### PR TITLE
[RF] Add Code-Squash AD support for RooFlexibleInterpVar.

### PR DESCRIPTION
**Pull request by grimmmyshini**
This PR adds Code-Squash AD support for RooFlexibleInterpVar.

### CODE REVIEW

File: roofit/roofitcore/inc/RooFit/Detail/EvaluateFuncs.h [hunk 1]
```diff
@@ -97,6 +97,74 @@ inline double poissonEvaluate(double x, double par)
...
+      if (x >= boundary) {
+         return total * std::pow(high / nominal, +paramVal);
+      } else if (x <= -boundary) {
+         return total * std::pow(low / nominal, -paramVal);
+      } else if (x != 0) {
+         return total * polyInterpValue(i, polCoeff, x);
+      }
+   }
+
+   return 0.0;
```
Comment (guitargeek) [inline roofit/roofitcore/inc/RooFit/Detail/EvaluateFuncs.h:165]:
```suggestion
   return total;
```
To fix the test failures and the problem that we sometimes get NaN

File: roofit/roofitcore/inc/RooFit/Detail/EvaluateFuncs.h [hunk 2]
```diff
@@ -97,6 +97,74 @@ inline double poissonEvaluate(double x, double par)
...
+      }
+   } else if (code == 4) {
+      double x = paramVal;
+      if (x >= boundary) {
+         return total * std::pow(high / nominal, +paramVal);
+      } else if (x <= -boundary) {
+         return total * std::pow(low / nominal, -paramVal);
+      } else if (x != 0) {
+         return total * polyInterpValue(i, polCoeff, x);
+      }
```
Comment (guitargeek) [inline roofit/roofitcore/inc/RooFit/Detail/EvaluateFuncs.h:162]:
```suggestion
    return total * polyInterpValue(i, polCoeff, x);
```
Do fix an ugly discontinuity that results in zero gradients, prompting minuit to do infinite parameter changes that result in NaN likelihood evaluations.

**Review by guitargeek — APPROVED** [review]
Hi Garima, thank you very much for implementing this! I have updated the PR with the implementation of the "hidden" `BatchMode("codegen")` option, a fix for constant parameters, and extending the `testHistFactory` unit test such that it covers the RooFlexibleInterpVar with AD.

**Comment by hahnjo** [comment]
Hello @guitargeek, as reported by Jenkins this seems to have broken `testHistFactory` on Windows. Could you please take a look?

**Pull request merged**